migration/src/m0000670_version_cmp/generic_version_matches.sql

```diff
@@ -0,0 +1,28 @@
+-- this is just an exact version match
+create or replace function generic_version_matches(version_p text, range_p version_range)
+    returns bool
+as
+$$
+begin
+    if range_p.low_version is not null then
+        if range_p.low_inclusive then
+            if version_p = range_p.low_version then
+                return true;
+            end if;
+        end if;
+    end if;
+
+    if range_p.high_version is not null then
+        if range_p.high_inclusive then
+            if version_p = range_p.high_version  then
+                return true;
+            end if;
+        end if;
+    end if;
+
+    return false;
+
+end
+$$
```

Comment:
you could rewrite to be a bit more compact:
 
```
create or replace function generic_version_matches(version_p text, range_p version_range)
    returns bool
as
$$
begin
    return (
        (range_p.low_version is not null and range_p.low_inclusive and version_p = range_p.low_version)
        or (range_p.high_version is not null and range_p.high_inclusive and version_p = range_p.high_version)
    );
end
$$ language plpgsql immutable;
```
untested,  I doubt this would result in such a speed up that it is worth reducing readability